src/lein_monolith/task/with_dependency_set.clj

```diff
@@ -0,0 +1,55 @@
+(ns lein-monolith.task.with-dependency-set
+  (:require
+    [clojure.java.io :as io]
+    [lein-monolith.plugin :as plugin]
+    [lein-monolith.task.util :as u]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]))
+
+
+(defn- maybe-remove-dependencies
+  [project opts]
+  (if (:only opts)
+    (assoc project :dependencies '())
+    project))
+
+
+(defn- reload-project
+  "Reloads the raw project.clj file for the given project.
+   This has the effect of removing any middleware that was added to the project
+   as well as any changes to the project map that were made in the current
+   session."
+  [project]
+  (-> project
+      (:root)
+      (io/file "project.clj")
+      (str)
+      (project/read-raw)))
+
+
+(defn- change-dependencies
+  [project dependency-set]
+  (let [[monolith _] (u/load-monolith! project)
+        dependencies (plugin/get-dependencies-from-set! monolith dependency-set project)]
```

Comment:
I think the error message may be a little misleading here, especially in the case where it's used from the metaproject. Either way, the dependency set name being used here isn't being "used from" the (sub)project, it's being passed on the command line.